addons/event_sale/models/sale_order.py

```diff
@@ -81,13 +81,23 @@ def _update_registrations(self, confirm=True, cancel_to_draft=False, registratio
             if cancel_to_draft:
                 existing_registrations.filtered(lambda self: self.state == 'cancel').action_set_draft()
 
+            first_registration = registration_data[0]['main_registration_id'] if registration_data else False
             for count in range(int(so_line.product_uom_qty) - len(existing_registrations)):
                 registration_vals = {}
                 if registration_data:
                     registration_vals = registration_data.pop()
                 # TDE CHECK: auto confirmation
                 registration_vals['sale_order_line_id'] = so_line.id
-                Registration.create(registration_vals)
+                if not first_registration:
+                    so_registrations = Registration.search([('sale_order_id', '=', so_line.order_id.id), ('event_id', '=', so_line.event_id.id)])
+                    if so_registrations.main_registration_id:
+                        first_registration = so_registrations.main_registration_id.id
+                    elif len(so_registrations) == 1:
+                        first_registration = so_registrations.id
```

Comment:
is it possible that main_registration_id differs among the returned so_registrations ?